File: CS/AuthorizationService/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace AuthorizationService.Controllers
{
    public class HomeController : Controller
    {
        [Authorize]
        public ActionResult Index()
        {
            return View();
        }

        public ActionResult About()
        {
            ViewData["Message"] = "Your application description page.";

            return View();
        }

        public ActionResult Contact()
        {
            ViewData["Message"] = "Your contact page.";

            return View();
        }
    }
}

File: VB/AuthorizationService/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace AuthorizationService.Controllers

    Public Class HomeController
        Inherits Controller

        <Authorize>
        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function About() As ActionResult
            ViewData("Message") = "Your application description page."
            Return View()
        End Function

        Public Function Contact() As ActionResult
            ViewData("Message") = "Your contact page."
            Return View()
        End Function
    End Class
End Namespace
